internal/uvm/network_test.go

```diff
@@ -0,0 +1,64 @@
+//go:build windows
+
+package uvm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Microsoft/hcsshim/internal/hns"
+)
+
+func Test_SortEndpoints(t *testing.T) {
+	type config struct {
+		endpointNames []string
+		targetName    string
+	}
+	tests := []config{
+		{
+			endpointNames: []string{"eth1", "eth0"},
+			targetName:    "eth0",
+		},
+		{
+			endpointNames: []string{"eth0", "eth1"},
+			targetName:    "eth0",
+		},
+		{
+			endpointNames: []string{"eth1", "name-eth0"},
+			targetName:    "name-eth0",
+		},
+		{
+			endpointNames: []string{"name-eth098", "name-eth0"},
+			targetName:    "name-eth0",
+		},
+		{
+			endpointNames: []string{"name-eth0", "name-eth098"},
+			targetName:    "name-eth0",
+		},
+		{
+			endpointNames: []string{"random-ifname", "another-random-ifname"},
+			// ordering shouldn't change so the first entry should still be first
```

Comment:
`slices.SortFunc` isn't stable, so if we need to relative order of equal elements to remain same we should use `slices.SortStableFunc`.

Reply:
Updated!